base/src/main/java/org/mozilla/jss/provider/java/security/KeyFactorySpi1_2.java

```diff
@@ -185,7 +185,7 @@ protected java.security.PrivateKey engineGeneratePrivate(KeySpec keySpec)
             // we need to chop off a leading zero byte
             if( y.bitLength() % 8 == 0 ) {
                 byte[] newBA = new byte[yBA.length-1];
-                assert(newBA.length >= 0);
+                assert(newBA.length > 0);
```

Comment:
This `assert` does nothing, as assertions only apply with debugging enabled in the JVM which is disabled by default. This should be replaced with proper `if/else` control flow.

Reply:
If the length is 0 the code here should work properly and an eventual exception will be generated later. I do not think it is necessary to modify the control flow at this level.

Reply:
OK, then the `assert` should be dropped completely because it does nothing but provide false assurance.

Reply:
Yeah, this particular `assert` is not very useful since it's already checked by Java, so I think we can just drop it.

Reply:
SonarCloud identify many issues with asserts overlapping the java checks. I think the idea was to stop the execution where the problem arise if executed in debug mode. The jave code and the assert are not doing exactly the same test. I would not drop this now but I am planning to evaluate these asserts again later.